plugins/test/framework.cc

```diff
@@ -45,7 +85,19 @@ proxy_wasm::WasmResult TestContext::log(uint32_t log_level,
   }
   return proxy_wasm::WasmResult::Ok;
 }
-
+proxy_wasm::BufferInterface* TestHttpContext::getBuffer(
+    proxy_wasm::WasmBufferType type) {
+  switch (type) {
+    case proxy_wasm::WasmBufferType::PluginConfiguration:
+      return TestContext::getBuffer(type);
+    case proxy_wasm::WasmBufferType::HttpRequestBody:
+      return &body_buffer_;
```

Comment:
nit: this should return nullptr if the body buffer type doesn't match the current callback, e.g. if the current callback is onRequestBody but getBuffer() is called with type=HttpResponseBody.

Reply:
I don't think we store that information in the TestContext.